src/ifcparse/IfcParse.cpp

```diff
@@ -1407,6 +1413,7 @@ void IfcEntityInstanceData::setArgument(size_t i, Argument* a, IfcUtil::Argument
 	}
 
 	attributes_[i] = copy;
+	delete a;
```

Comment:
Is it wise to elegant to delete `a` here, since it is just passed to this function? I see how it might be the quickest solution, but probably not the cleanest. I'd say at least add a code comment on this behaviour.

Reply:
you're right... all these arguments are created on the heap from client side. As a copy is made inside the function anyway, argument should be passed by const-ref instead of raw pointer. But that implies to change all call sites.

Reply:
I've investigated and found also why the python tests failed. The argument should not be deleted in the IfcEntiyInstanceData copy constructor (and it's the only case).
Instead of creating a copy and deleting the argument inside the function, I've added a parameter (default to false) to create a copy if needed (currently only in the IfcEntityInstanceData copy constructor) or directly set and use the argument pointer (the copy is not needed in this case).